analysers/analyser_osmosis_railway_angle.py

```diff
@@ -0,0 +1,111 @@
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+# This analyser is similar to Analyser_Osmosis_Way_Approximate
+# However, this one measures angles and not radiuses
+# It most likely indicates a node that was accidentaly dragged
+# While Analyser_Osmosis_Way_Approximate indicates that the way was drawn
+# with too few nodes
+
+from modules.OsmoseTranslation import T_
+from .Analyser_Osmosis import Analyser_Osmosis
+
+sql10 = """
+WITH
+    points AS (
+        SELECT
+            id,
+            tags->'{tag}' AS type,
+            (ST_DumpPoints(linestring)).geom AS geom4326,
+            (ST_DumpPoints(ST_Transform(linestring, {srid}))).geom AS geom,
+            (ST_DumpPoints(ST_Transform(linestring, {srid}))).path[1] AS index
+        FROM
+            {table_prefix}ways as ways
+        WHERE
+            tags != ''::hstore AND
+            tags?'{tag}' AND tags->'{tag}' IN ('{tag_values}') AND
+            ST_NPoints(linestring) >= 3
+    ),
+    angles AS (
+        SELECT
+            id, index, type,
+            lead(geom4326) OVER (PARTITION BY id ORDER BY index) AS geom4326,
+            lead(geom) OVER (PARTITION BY id ORDER BY index) AS geom,
+            degrees(
+                ST_Angle(geom,
+                     lead(geom, 1) OVER (PARTITION BY id ORDER BY index),
+                     lead(geom, 2) OVER (PARTITION BY id ORDER BY index)
+                )
+            ) angle,
+            ST_Distance(
+                geom,
+                lead(geom, 1) OVER (PARTITION BY id ORDER BY index)
+            ) AS length_1,
+            ST_Distance(
+                lead(geom, 1) OVER (PARTITION BY id ORDER BY index),
+                lead(geom, 2) OVER (PARTITION BY id ORDER BY index)
+            ) AS length_2
+        FROM
+            points
+    )
+
+    SELECT
+        id,
+        ST_AsText(geom4326),
+        angle,
+        type,
+        index
+    FROM
+        angles
+    WHERE
+        (angle < 150 OR angle > 210)
+        AND length_1 < 100
+        AND length_2 < 100
+"""
+
+class Analyser_Osmosis_Rail_Suspicious_Angle(Analyser_Osmosis):
+
+    def __init__(self, config, logger = None):
+        Analyser_Osmosis.__init__(self, config, logger)
+
+        self.classs_change[10] = self.def_class(item = 1190, level = 2, tags = ['geom', 'railway', 'fix:imagery'],
+            title = T_('Suspicious angle in railway'),
+            detail = T_(
+'''Sharp angles on a railway are suspicious. Maybe a node was accidentaly dragged'''),
+            fix = T_(
+'''After checking orthophotos, add nodes or move existing nodes.'''),
+            trap = T_(
+'''On service ways, train stations, train workshops that may be either a
```

Comment:
This is also in the other analyser and should probably be fixed in both, but I think the grammar of this line could be improved. Suggestion:
`On service ways, train stations and train workshops this could be a false positive.`